### PR TITLE
Sdl2 audio example

### PR DESCRIPTION
**Pull request by jonyluke**

### CODE REVIEW

**Comment by WinterMute** [comment]
SDL2 examples should use sdl2-config. See https://github.com/switchbrew/switch-examples/blob/master/graphics/sdl2/sdl2-simple/Makefile#L55 & https://github.com/switchbrew/switch-examples/blob/master/graphics/sdl2/sdl2-simple/Makefile#L62

Also: I think if we're going to add more sdl2 examples then we should probably move them all to an sdl2 folder rather than sub folders in other categories. Thoughts @fincs & @yellows8 ?

File: audio/sdl2-audio/Makefile [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+BUILD		:=	build
+SOURCES		:=	source
+DATA		:=	data
+INCLUDES	:=	include
+#ROMFS	:=	romfs
+
+#---------------------------------------------------------------------------------
+# options for code generation
+#---------------------------------------------------------------------------------
+ARCH	:=	-march=armv8-a -mtune=cortex-a57 -mtp=soft -fPIE
```
Comment (fincs) [inline audio/sdl2-audio/Makefile]: ARCH should be updated with the changes done in  d6db1e4

**Comment by fincs** [comment]
I would be fine either way.

File: audio/sdl2-audio/source/main.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        hidScanInput();
+
+        // hidKeysDown returns information about which buttons have been
+        // just pressed in this frame compared to the previous one
+        u64 kDown = hidKeysDown(CONTROLLER_P1_AUTO);
+
+        if (kDown & KEY_PLUS)
+            break; // break in order to return to hbmenu
+
+        if (kDown & KEY_A){
```
Comment (fincs) [inline audio/sdl2-audio/source/main.c]: Remove redundant `{}` braces

File: audio/sdl2-audio/source/main.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    // Free the loaded sound
+    Mix_FreeMusic(audio);
+	
+    // Shuts down SDL subsystems
+    SDL_Quit();
+	
+    // Deinitialize and clean up resources used by the console (important!)
+    consoleExit(NULL);
+    return 0;
+}
```
Comment (fincs) [inline audio/sdl2-audio/source/main.c]: Missing newline at EOF

**Comment by fincs** [comment]
Also another comment: This example should probably use romfs, and a small royalty free mp3 inside romfs; to make it work out of the box.

**Comment by jonyluke** [comment]
> SDL2 examples should use sdl2-config. See https://github.com/switchbrew/switch-examples/blob/master/graphics/sdl2/sdl2-simple/Makefile#L55 & https://github.com/switchbrew/switch-examples/blob/master/graphics/sdl2/sdl2-simple/Makefile#L62
> 
> Also: I think if we're going to add more sdl2 examples then we should probably move them all to an sdl2 folder rather than sub folders in other categories. Thoughts @fincs & @yellows8 ?

I've tried that, but compilation fails

**Comment by yellows8** [comment]
"compilation fails" What errors?

**Comment by jonyluke** [comment]
A few lines of `make`

```
linking sdl2-audio.elf
c:/devkitpro/devkita64/bin/../lib/gcc/aarch64-none-elf/8.3.0/../../../../aarch64-none-elf/bin/ld.exe: main.o: in function `main':
C:/Users/jony8/projects/sdl2-audio/source/main.c:26: undefined reference to `Mix_Init'
c:/devkitpro/devkita64/bin/../lib/gcc/aarch64-none-elf/8.3.0/../../../../aarch64-none-elf/bin/ld.exe: C:/Users/jony8/projects/sdl2-audio/source/main.c:30: undefined reference to `Mix_OpenAudio'
c:/devkitpro/devkita64/bin/../lib/gcc/aarch64-none-elf/8.3.0/../../../../aarch64-none-elf/bin/ld.exe: C:/Users/jony8/projects/sdl2-audio/source/main.c:33: undefined reference to `Mix_LoadMUS'
c:/devkitpro/devkita64/bin/../lib/gcc/aarch64-none-elf/8.3.0/../../../../aarch64-none-elf/bin/ld.exe: C:/Users/jony8/projects/sdl2-audio/source/main.c:49: undefined reference to `Mix_PlayMusic'
c:/devkitpro/devkita64/bin/../lib/gcc/aarch64-none-elf/8.3.0/../../../../aarch64-none-elf/bin/ld.exe: C:/Users/jony8/projects/sdl2-audio/source/main.c:56: undefined reference to `Mix_FreeMusic' 
```


```
collect2.exe: error: ld returned 1 exit status
make[1]: *** [/opt/devkitpro/libnx/switch_rules:80: /home/jony8/projects/sdl2-audio/sdl2-audio.elf] Error 1
make: *** [Makefile:165: build] Error 2
```

**Comment by jonyluke** [comment]
Now it works

**Comment by jonyluke** [comment]
Added romfs and a sound
Sound from https://freesound.org/people/jens.enk/sounds/434610/

File: audio/sdl2-audio/source/main.c [hunk 4]
```diff
@@ -17,7 +17,13 @@ int main(int argc, char* argv[])
 	
     consoleInit(NULL);
 
-    printf("Press A button to play the sound!\n");
+
+    Result rc = romfsInit();
+    if (R_FAILED(rc))
+    	printf("romfsInit: %08X\n", rc);
+	
+	else
```
Comment (yellows8) [inline audio/sdl2-audio/source/main.c]: tabs->spaces, same below.

File: audio/sdl2-audio/source/main.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	 
+    // Load support for the MP3 format 
+    Mix_Init(MIX_INIT_MP3);
+	 
+    // open 44.1KHz, signed 16bit, system byte order,
+    //  stereo audio, using 4096 byte chunks
+    Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, MIX_DEFAULT_CHANNELS, 4096);
+	
+    // Load sound file to use
+	//Sound from https://freesound.org/people/jens.enk/sounds/434610/
```
Comment (fincs) [inline audio/sdl2-audio/source/main.c]: tab -> 4 spaces.

Comment (jonyluke) [reply audio/sdl2-audio/source/main.c]: I know, I'm using Notepad++ where the spaces appear correct, I'd better switch to visual code

File: audio/sdl2-audio/source/main.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        consoleUpdate(NULL);
+    }
+
+    // Free the loaded sound
+    Mix_FreeMusic(audio);
+
+    // Shuts down SDL subsystems
+    SDL_Quit();
+
+    // Deinitialize and clean up resources used by the console (important!)
```
Comment (fincs) [inline audio/sdl2-audio/source/main.c]: Missing romfsExit

File: audio/sdl2-audio/source/main.c [hunk 7]
```diff
@@ -64,6 +64,8 @@ int main(int argc, char *argv[])
     // Shuts down SDL subsystems
     SDL_Quit();
 
+    romfsExit();
```
Comment (WinterMute) [inline audio/sdl2-audio/source/main.c]: Slightly nitpicky but can you move this down below the Deinitialize comment - it is libnx deinitialization code after all.

**Pull request merged**